packages/shared/refThrottled/index.ts

```diff
@@ -10,18 +14,31 @@ import { useThrottleFn } from '../useThrottleFn'
  * @param  delay  A zero-or-greater delay in milliseconds. For event callbacks, values around 100 or 250 (or even higher) are most useful.
  * @param [trailing] if true, update the value again after the delay time is up
  * @param [leading] if true, update the value on the leading edge of the ms timeout
+ * @param [cloneHandler] By default, it use `JSON.parse(JSON.stringify(value))` to clone
  */
-export function refThrottled<T>(value: Ref<T>, delay = 200, trailing = true, leading = true) {
+export function refThrottled<T>(value: Ref<T>, delay = 200, trailing = true, leading = true, cloneHandler = cloneFnJSON) {
   if (delay <= 0)
     return value
+  const throttled = ref() as Ref<T>
+  const isEqualityClone = cloneHandler === cloneFnJSON
+  const setThrottled = () => {
+    try {
+      throttled.value = isReferenceType(value.value) ? cloneHandler(value.value) : value.value
+    }
+    catch (error) {
+      if (isEqualityClone)
```

Comment:
need a brackets here

Reply:
Do I need an effectScope here? Like this
```typescript
  const updater = useThrottleFn(setThrottled, delay, trailing, leading)
  const scope = effectScope(true)
  scope.run(() => {
    watch(value, updater, {
      deep: isReferenceType(value.value),
    })
  })

  tryOnScopeDispose(() => {
    scope.stop()
  })
```

Reply:
No, if the user uses the hook in the setup function, it will be disposed automatically